tst/EnergyPlus/unit/MixedAir.unit.cc

```diff
@@ -5486,32 +5500,95 @@ TEST_F(EnergyPlusFixture, MechVentController_VRPNoCap)
+    int primaryOutletNode2 = state->dataZoneEquip->ZoneEquipConfig(2).AirDistUnitCool(1).OutNode; // primary node of zone terminal unit
+    state->dataLoopNodes->Node(primaryOutletNode2).Temp = 15.0;
+    state->dataLoopNodes->Node(primaryOutletNode2).HumRat = 0.001;
+    state->dataLoopNodes->Node(primaryOutletNode2).MassFlowRate = SupplyMassFlow2;
+
+    state->dataMixedAir->VentilationMechanical(1).SysDesOA = DesignOAMassFlow; // Set design outdoor air flow rate
+    OAMassFlow = state->dataMixedAir->VentilationMechanical(1).CalcMechVentController(*state, SysMassFlow);
+
+    EXPECT_TRUE(OAMassFlow > DesignOAMassFlow); // Expect that the system OA is greater than the design OA air flow
+    EXPECT_TRUE(OAMassFlow < SysMassFlow);      // Expect that the system OA is less than the system air flow
+
+    // Extend test to cover the Max OA method
+    state->dataSize->OARequirements(1).OAFlowMethod = DataSizing::OAFlowCalcMethod::Max;
+    DesignOAMassFlow1 =
+        state->dataEnvrn->StdRhoAir * max(state->dataHeatBal->Zone(1).TotOccupants * 0.00944, state->dataHeatBal->Zone(1).FloorArea * 0.0003048006);
+    DesignOAMassFlow2 =
+        state->dataEnvrn->StdRhoAir * max(state->dataHeatBal->Zone(2).TotOccupants * 0.00944, state->dataHeatBal->Zone(2).FloorArea * 0.0003048006);
+    DesignOAMassFlow = DesignOAMassFlow1 + DesignOAMassFlow2;
+    SupplyMassFlow1 = DesignOAMassFlow1 / 0.5;
+    SupplyMassFlow2 = DesignOAMassFlow2 / 0.2;
+    state->dataLoopNodes->Node(primaryInletNode1).MassFlowRate = SupplyMassFlow1;
+    state->dataLoopNodes->Node(primaryInletNode2).MassFlowRate = SupplyMassFlow2;
+    state->dataLoopNodes->Node(primaryOutletNode1).MassFlowRate = SupplyMassFlow1;
+    state->dataLoopNodes->Node(primaryOutletNode2).MassFlowRate = SupplyMassFlow2;
+    SysMassFlow = SupplyMassFlow1 + SupplyMassFlow2;                           // System supply mass flow rate [kg/s]
+    state->dataMixedAir->VentilationMechanical(1).SysDesOA = DesignOAMassFlow; // Set design outdoor air flow rate
+    OAMassFlow = state->dataMixedAir->VentilationMechanical(1).CalcMechVentController(*state, SysMassFlow);
+
+    EXPECT_TRUE(OAMassFlow > DesignOAMassFlow); // Expect that the system OA is greater than the design OA air flow
+    EXPECT_TRUE(OAMassFlow < SysMassFlow);      // Expect that the system OA is less than the system air flow
+
+    // Starve one zone so that OAMassFlow = SysMassFlow
+    SupplyMassFlow1 = DesignOAMassFlow1 * 0.5;
+    SupplyMassFlow2 = DesignOAMassFlow2 / 0.2;
+    state->dataLoopNodes->Node(primaryInletNode1).MassFlowRate = SupplyMassFlow1;
+    state->dataLoopNodes->Node(primaryInletNode2).MassFlowRate = SupplyMassFlow2;
+    state->dataLoopNodes->Node(primaryOutletNode1).MassFlowRate = SupplyMassFlow1;
+    state->dataLoopNodes->Node(primaryOutletNode2).MassFlowRate = SupplyMassFlow2;
+    SysMassFlow = SupplyMassFlow1 + SupplyMassFlow2;                           // System supply mass flow rate [kg/s]
```

Comment:
Not sure of the need for code duplication here, lines 5565-5572 and 5580-5585. I don't think those would have changed from previous test. You lost me on the "Starve one zone" comment

Reply:
Oh, I see, before SupplyMassFlow1 was /0.5 and now it's *0.5. That makes sense. So the "1" vars changed.